File: figures/05-dynamic-programming/snippet-max-subarray-brute-force.java
    public static Result maxSubarrayBruteForce(long[] a) {
        if (a == null || a.length == 0) throw new IllegalArgumentException();
        Result best = new Result(a[0], 0, 0);
        for (int i = 0; i < a.length; i++) {
            long sum = 0;
            for (int j = i; j < a.length; j++) {
                sum = Math.addExact(sum, a[j]);
                if (sum > best.sum()) best = new Result(sum, i, j);
            }
        }
        return best;
    }

File: figures/05-dynamic-programming/snippet-max-subarray-brute-force.c
static bool max_subarray_brute_force(const long long *a, size_t n, MaxSubarrayResult *out) {
    if (!a || !out || n == 0) return false;
    MaxSubarrayResult best = {a[0], 0, 0};
    for (size_t i = 0; i < n; ++i) {
        long long sum = 0;
        for (size_t j = i; j < n; ++j) {
            sum += a[j];
            if (sum > best.sum) { best.sum = sum; best.start = i; best.end = j; }
        }
    }
    *out = best;
    return true;
}
